Tools/CISupport/ews-build/steps.py

```diff
@@ -4823,3 +4847,103 @@ def doStepIf(self, step):
+    name = 'push-pull-request-branch'
+    haltOnFailure = True
+
+    def __init__(self, **kwargs):
+        super(PushPullRequestBranch, self).__init__(logEnviron=False, timeout=300, **kwargs)
+
+    def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
+        remote = self.getProperty('github.head.repo.full_name').split('/')[0]
+        head_ref = self.getProperty('github.head.ref')
+        self.command = ['git', 'push', remote, head_ref, '-f']
+
+        username, access_token = GitHub.credentials()
+        self.workerEnvironment['GIT_USER'] = username
+        self.workerEnvironment['GIT_PASSWORD'] = access_token
+
+        return super(PushPullRequestBranch, self).start()
+
+    def getResultSummary(self):
+        if self.results == SKIPPED:
+            return {'step': 'No pull request branch to push to'}
+        if self.results != SUCCESS:
+            return {'step': 'Failed to push to pull request branch'}
+        return {'step': 'Pushed to pull request branch'}
+
+    def doStepIf(self, step):
+        return CURRENT_HOSTNAME == EWS_BUILD_HOSTNAME and self.getProperty('github.number') and self.getProperty('github.head.ref') and self.getProperty('github.head.repo.full_name')
+
+    def hideStepIf(self, results, step):
+        return not self.doStepIf(step)
+
+class UpdatePullRequest(shell.ShellCommand, GitHubMixin):
+    name = 'update-pull-request'
+    haltOnFailure = True
+    command = ['git', 'log', '-1', '--no-decorate']
+    ESCAPE_TABLE = {
+        '"': '&quot;',
+        "'": '&apos;',
+        '>': ' &gt;',
+        '<': '&lt;',
+        '&': '&amp;',
+    }
+
+    @classmethod
+    def escape_html(cls, message):
+        message = ''.join(cls.ESCAPE_TABLE.get(c, c) for c in message)
+        return re.sub(r'(https?://[^\s<>,:;]+)', r'<a href="\1">\1</a>', message)
+
+    def __init__(self, **kwargs):
+        super(UpdatePullRequest, self).__init__(logEnviron=False, timeout=300, **kwargs)
+
+    @defer.inlineCallbacks
+    def _addToLog(self, logName, message):
+        try:
+            log = self.getLog(logName)
+        except KeyError:
+            log = yield self.addLog(logName)
+        log.addStdout(message)
+
+    def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
+        self.log_observer = BufferLogObserverClass(wantStderr=True)
+        self.addLogObserver('stdio', self.log_observer)
+
+        return super(UpdatePullRequest, self).start()
+
+    def getResultSummary(self):
+        if self.results == SKIPPED:
+            return {'step': 'No pull request to update'}
+        if self.results != SUCCESS:
+            return {'step': 'Failed to update pull request'}
+        return {'step': 'Updated pull request'}
+
+    def evaluateCommand(self, cmd):
+        rc = super(UpdatePullRequest, self).evaluateCommand(cmd)
+
+        loglines = self.log_observer.getStdout().splitlines()
+
+        title = loglines[4][4:].rstrip()
```

Comment:
why 4?

Reply:
We're extracting text from `git log -1 --no-decorate`, looks like this:
```
commit d758f169aac1945c71ec21583150d4da88d0cea9
Author: Devin Rousso <drousso@apple.com>
Date:   Wed Mar 30 17:01:11 2022 +0000

    Unreviewed, fix test after r285521 and r271735
    
    * http/tests/paymentrequest/updateWith-shippingOptions.https.html:
    Remove a duplicate `assert_equals` added in r271735 (which should've been removed in r285521).
    Drive-by: Also check that the `selected` of each provided shipping method/option matches.
    
    
    Canonical link: https://commits.webkit.org/249027@main
    git-svn-id: https://svn.webkit.org/repository/webkit/trunk@292107 268f45cc-cd09-0410-ab3c-d52691b4dbfc
```